test/e2e/authorization.go

```diff
@@ -0,0 +1,572 @@
+				tc.data.resources = []string{
+					"namespaces",
+					"nodes",
+					"rbac.authorization.k8s.io/clusterroles",
+					"storage.k8s.io/storageclasses",
+					"apiextensions.k8s.io/customresourcedefinitions",
+				}
+				g.It("should allow read access", func() {
+					tc.data.verbs = []string{"get", "list", "watch"}
+					tc.run(context.TODO(), cs, true)
+					gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+				})
+				g.It("should deny write access", func() {
+					tc.data.verbs = []string{"create", "update", "delete", "patch"}
+					tc.run(context.TODO(), cs, false)
+					gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+				})
+			})
+		})
+	})
+
+	g.Context("For PowerUser, Manual and Emergency groups", func() {
+		var tc testCase
+		g.BeforeEach(func() {
+			tc.data.groups = [][]string{
+				{"PowerUser"},
+				{"Manual"},
+				{"Emergency"},
+			}
+			tc.data.users = []string{"test-user"}
+		})
+
+		g.It("should deny read access to Secrets in kube-system and visibility namespaces", func() {
+			tc.data.resources = []string{"secrets"}
+			tc.data.namespaces = []string{"kube-system", "visibility"}
+			tc.data.verbs = []string{"get", "list", "watch"}
+			tc.run(context.TODO(), cs, false)
+			gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+		})
+
+		g.It("should deny write access to Nodes", func() {
+			tc.data.resources = []string{"nodes"}
+			tc.data.verbs = []string{"create", "update", "delete", "patch"}
+			tc.run(context.TODO(), cs, false)
+			gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+		})
+
+		g.It("should deny write access to DaemonSets", func() {
+			tc.data.resources = []string{"apps/daemonsets"}
+			tc.data.verbs = []string{"create", "update", "delete", "patch"}
+			tc.run(context.TODO(), cs, false)
+			gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+		})
+
+		g.It("should allow deleting CRDs", func() {
+			tc.data.resources = []string{"apiextensions.k8s.io/customresourcedefinitions"}
+			tc.data.verbs = []string{"delete"}
+			tc.run(context.TODO(), cs, true)
+			gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+		})
+
+		g.It("should deny deleting kube-system or visibility namespaces", func() {
+			tc.data.resources = []string{"namespaces"}
+			tc.data.namespaces = []string{"kube-system", "visibility"}
+			tc.data.verbs = []string{"delete"}
+			tc.run(context.TODO(), cs, false)
+			gomega.Expect(tc.output.passed).To(gomega.BeTrue(), tc.output.String())
+		})
+
+		g.When("the resource is a namespaced resource", func() {
+			g.BeforeEach(func() {
+				tc.data.resources = []string{
+					"pods",
+					"apps/deployments",
+					"apps/statefulsets",
+					"apps/deployments/scale",
+					"apps/statefulsets/scale",
+					"services",
+					"persistentvolumes",
+					"persistentvolumeclaims",
+					"configmaps",
+				}
+				tc.data.verbs = []string{"create", "update", "delete", "patch"}
```

Comment:
We could extract these "static" lists as vars similar to `allGroups` in the top and then re-use them. For example:

```go
	var (
		namespacedResource = []string{
			"pods",
			"apps/deployments",
			"apps/statefulsets",
			"apps/deployments/scale",
			"apps/statefulsets/scale",
			"services",
			"persistentvolumes",
			"persistentvolumeclaims",
			"configmaps",
		}
		globalResources = []string{
			"namespaces",
			"nodes",
			"rbac.authorization.k8s.io/clusterroles",
			"storage.k8s.io/storageclasses",
			"apiextensions.k8s.io/customresourcedefinitions",
		}
		readOperations  = []string{"get", "list", "watch"}
		writeOperations = []string{"create", "update", "delete", "patch"}
	)
```

Reply:
That's actually a good suggestions. I'll do that. Thanks.

Reply:
Added in this commit: https://github.com/zalando-incubator/kubernetes-on-aws/pull/8536/commits/1b13defef78ff9c53dbdc6c43a9e14659f868991

Looks much better 🙂